src/LuaTransformer.ts

```diff
@@ -3267,17 +3304,66 @@ export class LuaTransformer {
         return tstl.createBinaryExpression(expression, tstl.createNumericLiteral(1), tstl.SyntaxKind.AdditionOperator);
     }
 
+    protected findScope(...scopeTypes: ScopeType[]): Scope | undefined {
```

Comment:
Might be interesting to use a bitmask for this instead of an array.

Reply:
that's a good idea!